### PR TITLE
feat(merchant-migration): decide and perform a subscription cutover

### PR DESCRIPTION
**Pull request by psincraian**
## Summary

**Related Issue**: n/a

> 📚 Stack 5/7 · base is #13625. Reviewable now; merge after its base. **This is the one to read slowly** — it decides whether real money moves.

The cutover decision engine, plus the subscription-side transition it drives. **Nothing schedules it yet.**

## What

- `subscription_service.activate_imported(...)` — hand billing of a paused imported subscription over to Polar.
- `SubscriptionCutover.run(record)` — decide whether one subscription may move, and move it. Returns `moved` / `skipped(reason)` / `failed(reason)`.

## Why

Cutting a subscription over is the only irreversible act in a migration: it cancels a live subscription on the merchant's own Stripe account. Everything that could go wrong has to be checked before that, and anything unclear has to leave the subscription where it is rather than guess.

`activate_imported` exists because `resume` is the wrong shape here. Resuming starts a fresh period and charges immediately; the customer already paid the old provider through the end of the current period, so Polar has to pick the cycle up exactly there.

## How

**Order of checks, all read-only, cancellation last.** Still live on the source, on the same price, single line item, no coupon, still charged automatically, not already set to cancel · renewal more than 24h out · copied card confirms a zero-amount `SetupIntent`. Only then stop the source and activate. A subscription that fails any check keeps billing on the source, stays paused on Polar, and carries a reason the merchant can act on.

**The period is read at cutover, not at import.** Weeks of source renewals have happened in between, so the staged period is stale; activating on it would bill the wrong date.

**24h window.** A renewal landing sooner risks the source charging the very period we're taking over. Skipping is retryable: once the source has charged, the next period opens up.

**`SetupIntent`, cards only.** A zero-amount confirmation is a real validity probe for a card and tells us whether it can be charged off-session, which is what the first renewal will do. It says nothing useful about a bank debit and confirming one needs a mandate the copy doesn't carry, so those are accepted as-is.

**Already stopped by us → skip the checks and finish.** That's the crash-retry path from #13624 landing here: re-running the checks against our own cancellation would read as churn.

**No resumed email, no order.** From the customer's side nothing paused and nothing is changing today; the first they should notice is the renewal they already expected, charged by Polar.

Edge cases covered by tests: source gone · cancelled after import · `past_due` · already ending · renewal too soon · renewal in the past · plan changed · second line item · manual invoicing · no card yet · card declined · card needs SCA · Polar subscription already active (idempotent) or cancelled by hand · customer deleted · Stripe unreachable (retryable, not skipped).

## Checklist

- [x] This PR addresses a single concern (one bug fix, one feature, one refactor)
- [x] The diff is reasonably sized and easy to review
- [x] New functionality is covered by tests — 20 cases on the engine, and every skip asserts the source was **not** cancelled
- [x] Linting and type checking pass (`uv run task lint && uv run task lint_types`)
- [x] No unrelated changes or drive-by fixes are included

---
_Generated by [Claude Code](https://claude.ai/code/session_011PzjzyWCvUEeXWuG4UGy8R)_

<a href="https://cubic.dev/pr/polarsource/polar/pull/13626?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: server/tests/subscription/test_service.py [hunk 1]
```diff
@@ -3132,6 +3132,85 @@ async def test_valid(
...
+        assert updated.paused_at is None
+        assert updated.current_period_start == period_start
+        assert updated.current_period_end == period_end
+        assert updated.payment_method_id == payment_method.id
+        enqueue_benefits_grants_mock.assert_called_once_with(session, updated)
+        # No order: the customer is not charged for a period they already paid.
+        enqueued = [call.args[0] for call in enqueue_job_mock.call_args_list]
+        assert "order.create_subscription_order" not in enqueued
+
+    async def test_does_not_tell_the_customer_anything_resumed(
```
[the remark this answers is not in the document]
Comment (psincraian) [reply server/tests/subscription/test_service.py:3418]: Not applicable — the premise doesn't hold here. `tests/fixtures/worker.py` sets the context for every test:

```python
@pytest.fixture(autouse=True)
def set_job_queue_manager_context() -> None:
    _job_queue_manager.set(JobQueueManager())
```

So `JobQueueManager.get()` never raises in this suite, and jobs land in a per-test in-memory manager rather than a real queue. This test passes as written.

The sibling test does take `enqueue_job_mock`, because it asserts on what was enqueued. This one asserts that no resumed hook fired, so pulling the fixture in would add a parameter it never reads — and that's the more common shape in this file (`TestResume::test_not_paused`, among others).

Leaving as-is.

---
_Generated by [Claude Code](https://claude.ai/code)_
